docs/src/idiomatic/assignments.md

```diff
@@ -0,0 +1,129 @@
+On the other hand, if we change the example to
+```tla
+/\ y' = 3
+/\ \/ y = 1
+   \/ y' = 2
+```
+the only assignment has to be `y' = 3`. While one of the disjuncts is an assignment candidate and the other is not, the balance requirement is not violated here, since neither disjunct is chosen as an assignment.
+
+Similar rules apply to if-then-else: both the `THEN _` and `ELSE _` branch must assign the same variables, however, the `IF _` condition is ignored when determining assignments.
+
+### Assignment-free Expressions
+Not all expressions may contain assignments. Given a transition operator `A`, based on the shape of `A`, the following holds:
+- If `A` has the shape `A_1 /\ ... /\ A_n`, then assignments are selected from `A_1, ... , A_n` sequentially, subject to the syntax-order rule.
+- If `A` has the shape `A_1 \/ ... \/ A_n`, then assignments are selected in all `A_1, ... , A_n` independently, subject to the balance rule.
+- If `A` has the shape `IF p THEN A_1 ELSE A_2`, then:
+    - `p` may not contain assignments. Any assignment candidates in `p` are subject to the assignment-before-use rule.
+    - Assignments are selected in both `A_1` and `A_n` independently, subject to the balance rule.
+- If `A` has the shape `\E x \in S: A_1`, then:
+    - `S` may not contain assignments. Any assignment candidates in `S` are subject to the assignment-before-use rule.
+    - Assignments are selected in `A_1`
+- In any other case, `A` may not contain assignments, however, any assignment candidates in `A` are subject to the assignment-before-use rule.
+
+Examples:
+```tla
+A == /\ x' = 2 
+     /\ \E s \in { t \in 1..10 : x' > t }: y' = s
+
+B == \E s \in { t \in 1..10 : x' > t }: y' = s
+
+C == /\ x' = 1 
+     /\ IF x' = 0 /\ 2 \in {x', x' + 2, 0}
+        THEN y' = 1 
+        ELSE y' = 2
+
+D == IF x' = 0 
+     THEN y' = 1 
+     ELSE y' = 2
+
+E == /\ x' = 2 
+     /\ \A s \in { t \in 1..10 : x' > t }: y' = s
+```
+Operator `A` contains assignments to both `x` and `y`; while `x' > t` uses `x'`, it does not violate the assignment-before-use rule, since the assignment to `x` precedes the expression, w.r.t. syntax order.
+In operator `B`, the assignment to `x` is missing, therefore `x' > t` produces an exception, as it violates assignment-before-use.
+The case in `C` is similar to `A`; conditions of the if-then-else operator may not contain assignments to `x`, so `x' = 0` can never be one, but they may use `x'`, since a preceding expression (`x' = 1`) qualifies as an assignment.
+The operator `D` produces an exception, for the same reason as `B`; even though `x' = 0` is an assignment candidate, if-conditions are assignment-free, so `x' = 0` cannot be chosen as an assignment to `x`.
+Lastly, while `E` looks almost identical to `A`, the key difference is that expressions under universal quantifiers may not contain assignments. Therefore, `y' = s` is *not* an assignment to `y` and thus violates assignment-before-use.
```

Comment:
I think this would be easier to parse if you interleave the examples and the analysis, and and clearly flag in advance which examples are meant to be valid and which invalid, likely with an inline comment. So like

> Examples:
>
> ```tla
> A == /\ x' = 2 
>     /\ \E s \in { t \in 1..10 : x' > t }: y' = s
> ```
>
> Operator `A` contains assignments to both `x` and `y`; while `x' > t` uses `x'`, it does not violate the assignment-before-use rule, since the assignment to `x` precedes the expression, w.r.t. syntax order.
> 
> ```tla
> (* INVALID *)
> B == \E s \in { t \in 1..10 : x' > t }: y' = s
> ```
> 
> In operator `B`, the assignment to `x` is missing, therefore `x' > t` produces an exception, as it violates assignment-before-use.

Reply:
Makes sense